Customize how hubot-grafana connects to S3 endpoints

GrandPubba opened the pull request:
Customize how hubot-grafana connects to S3 endpoints by adding the following
environment variables:

HUBOT_GRAFANA_S3_PORT - Port of the S3 endpoint
HUBOT_GRAFANA_S3_STYLE - Bucket style of the S3 endpoing ( 'virtualHosted' or 'path' )

stephenyeargin commented on src/grafana.coffee:
```diff
@@ -274,10 +289,18 @@ module.exports = (robot) ->
+
     req = client.put(filename, headers)
+
+    imageUrl = 
```
Looks to be a stray line?

stephenyeargin commented on src/grafana.coffee:
```diff
@@ -44,7 +44,15 @@ module.exports = (robot) ->
+    s3_port = process.env.HUBOT_GRAFANA_S3_PORT if process.env.HUBOT_GRAFANA_S3_PORT
+  else
+    s3_region = 'us-standard'
+    s3_region = process.env.HUBOT_GRAFANA_S3_REGION if process.env.HUBOT_GRAFANA_S3_REGION
```
`s3_region = process.env.HUBOT_GRAFANA_S3_REGION or 'us-standard'` achieves the same thing as the two lines above. Came with #34 to be more CoffeeScript-ish.

stephenyeargin commented on src/grafana.coffee:
```diff
@@ -44,7 +44,15 @@ module.exports = (robot) ->
+  s3_style = process.env.HUBOT_GRAFANA_S3_STYLE if process.env.HUBOT_GRAFANA_S3_STYLE
+
+  if process.env.HUBOT_GRAFANA_S3_ENDPOINT
+    s3_region = undefined
```
None of the alternative providers use the region flag for anything interesting?

stephenyeargin commented on test/grafana-s3-test.coffee:
```diff
@@ -0,0 +1,124 @@
+Helper = require('hubot-test-helper')
```
:heart_eyes_cat: for adding test coverage to the S3 integration!

stephenyeargin commented on test/grafana-test.coffee:
```diff
@@ -25,6 +25,7 @@ describe 'grafana', ->
   afterEach ->
     room.destroy()
     nock.cleanAll()
+    delete process.env.HUBOT_GRAFANA_HOST
```
Whoops, forgot that bit. :sweat_smile:

stephenyeargin commented:
You don't have to close the PR, should be able to push to the same branch it :octocat: takes care of updates.

GrandPubba commented:
Sorry about that.  Thought rewriting commits was confusing things but it wasn't.  In any case the most recent version I believe addresses your concerns.

stephenyeargin commented:
No worries! I think this looks good to go! :+1:

stephenyeargin commented:
Released with `v1.4.8`
